code/script.js

```diff
@@ -222,104 +225,185 @@ const generateBoard = () => {
-  */
+      // Hair
+  else if (category === 'hair') {
+    if (keep) {
+      alert(`That's correct, the person has ${value} hair! Keep all people with ${value} hair`)
+      charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
+    } else {
+      alert(`Nope, the person doesn't have ${value} hair! Remove all people with ${value} hair`)
+      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
+    }
+  }
 
-  // Invoke a function to redraw the board with the remaining people.
+
+  // Invokes the function to redraw the board with the remaining people.
+  generateBoard()
 }
 
-// when clicking guess, the player first have to confirm that they want to make a guess.
+// When clicking guess, the player first have to confirm that they want to make a guess.
+// This stores the interaction from the player in a variable.
+  // Asks the player to confirm
+  // If the player wants to guess, the checkMyGuess function invokes.
 const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+  const result = window.confirm(`Are you sure this is your choice?`);
+
+  if (result) {
+    checkMyGuess(personToConfirm);
+  }
 }
 
-// If you confirm, this function is invoked
+// If player confirms, this function is invoked
 const checkMyGuess = (personToCheck) => {
-  // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
+  
+  // 1. Checks if the personToCheck is the same as the secret person's name
+  // 2. Sets a Message to show in the win or lose section accordingly
+  // 3. Shows the win or lose section
+  // 4. Hides the game board
+  if (personToCheck === secret.name) {
+    pauseBackgroundMusic()
+    alert("We have a winner! Well done!");
+    winOrLose.style.display='block';
+  }
+  else {
+    alert("Computer says no. Game over!");
+    winOrLose.style.display='block';
+    pauseBackgroundMusic()
+  }
 }
 
 // Invokes the start function when website is loaded
 start()
 
+// This function runs the background music
+const playBackgroundMusic = () => {
+  backgroundMusic.play()
+}
+
+// This function pauses the background music
+const pauseBackgroundMusic = () => {
+  backgroundMusic.pause()
+}
+
 // All the event listeners
 restartButton.addEventListener('click', start)
+questions.addEventListener('change', selectQuestion)
+questions.addEventListener('change', playBackgroundMusic) //Music starts when the player clicks on the dropdown list
+findOutButton.addEventListener('click', checkQuestion)
+playAgainButton.addEventListener('click', () => {
+  start()
+  winOrLose.style.display='none'; //Invokes the Play again-button and displays the game board
+})
+
+
```

Comment:
I tried out the game and all your functions are working as they should. Really liked that you added nice background music to the game! 
The code is very structured with great comments on what's happening on each step. All blue requirements are met. Well done Karoline!!! :)